eth/downloader/beaconsync.go

```diff
@@ -201,8 +201,6 @@ func (d *Downloader) findBeaconAncestor() (uint64, error) {
 		chainHead = d.blockchain.CurrentBlock()
 	case SnapSync:
 		chainHead = d.blockchain.CurrentSnapBlock()
-	default:
-		chainHead = d.lightchain.CurrentHeader()
```

Comment:
You need a new default, even if an error clause

Reply:
Note, if this is internal and you already validated that only the first two can happen, then use a panic as the default. Anything really to make it future proof.

Reply:
panic it is then.  The sync mode is only ever set with the result of `Ethereum.SyncMode` in `eth/backend.go`, and it can only return snap or full sync.